Tests/SonarScanner.MSBuild.PreProcessor.Test/ProcessedArgsTests.cs

```diff
@@ -361,12 +390,86 @@ public void ProcArgs_OperatingSystem_AutoDetection(PlatformOS platformOS, string
             operatingSystemProvider.OperatingSystem().Returns(_ => platformOS);
             var sut = CreateDefaultArgs(operatingSystemProvider: operatingSystemProvider);
             sut.OperatingSystem.Should().Be(expectedOperatingSystem);
+            sut.IsValid.Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void ProcArgs_UserHome_ParameterProvided()
+        {
+            var directoryWrapper = Substitute.For<IDirectoryWrapper>();
+            directoryWrapper.Exists(@"C:\Users\user\.sonar").Returns(true);
+            var sut = CreateDefaultArgs(new ListPropertiesProvider([new Property(SonarProperties.UserHome, @"C:\Users\user\.sonar")]), directoryWrapper: directoryWrapper);
+            sut.UserHome.Should().Be(@"C:\Users\user\.sonar");
+            sut.IsValid.Should().BeTrue();
+            logger.AssertNoErrorsLogged();
+            logger.AssertNoWarningsLogged();
+        }
+
+        [TestMethod]
+        public void ProcArgs_UserHome_ParameterProvided_DoesNotExists()
+        {
+            var directoryWrapper = Substitute.For<IDirectoryWrapper>();
+            directoryWrapper.Exists(@"C:\Users\user\.sonar").Returns(false);
+            var sut = CreateDefaultArgs(new ListPropertiesProvider([new Property(SonarProperties.UserHome, @"C:\Users\user\.sonar")]), directoryWrapper: directoryWrapper);
+            sut.UserHome.Should().BeNull();
+            sut.IsValid.Should().BeFalse();
+            logger.AssertErrorLogged(@"The provided value for 'sonar.userHome' 'C:\Users\user\.sonar' does not exists. Specify a valid directory for 'sonar.userHome'.");
```

Comment:
I see we log an error, not a warning. Do we want to stop execution if the path does not exist or to fall back to current behavior?

Reply:
Good question. The behavior is undefined in the spec.

Reply:
I asked the analysis experience squad:
https://sonarsource.slack.com/archives/C06J6RWRPMX/p1720015489072249

Reply:
Feedback from Julien via mail from July 8 "JRE provisioning questions":
> 3) Error handling is a part that has not been properly specified. In the past, the guiding principles were to fail fast, but I think nowadays we want to be as resilient as possible to ensure the best user experience.
To be honest, I don't know how the scanner CLI behaves, but if I had to guess, it would likely stop with a beautiful Java stack trace (/s) as soon as an IO operation fails.
I suggest you do what is simpler for you while keeping user experience in mind. If you have a doubt, maybe double-check with Alexandre Gigleux.  That would likely lead to improving the spec, and aligning all other scanners.